src/fabric/src/fabric2_db.erl

```diff
@@ -969,6 +970,145 @@ fold_docs(Db, UserFun, UserAcc0, Options) ->
     end).
 
 
+fold_docs(Db, DocIds, UserFun, UserAcc0, Options) ->
+    fabric2_fdb:transactional(Db, fun(TxDb) ->
+        try
+            NeedsTreeOpts = [revs_info, conflicts, deleted_conflicts],
+            NeedsTree = (Options -- NeedsTreeOpts /= Options),
+
+            FetchRevs = case NeedsTree of
+                true ->
+                    fun(DocId) ->
+                        fabric2_fdb:get_all_revs_future(TxDb, DocId)
```

Comment:
I don't think you've included an implementation of `get_all_revs_future` in fabric2_fdb unless I'm really blind. I'd probably add it as part of the refactoring commit.

Reply:
Good spot. Fixed